src/wasm-lib/kcl/src/ast/types.rs

```diff
@@ -2753,6 +2753,41 @@ async fn execute_pipe_body(
     }
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, JsonSchema, Bake, FromStr, Display)]
+#[databake(path = kcl_lib::ast::types)]
+#[serde(tag = "type")]
+#[display(style = "snake_case")]
+pub enum FnArgPrimitive {
+    /// A string type.
+    String,
+    /// A number type.
+    Number,
+    /// A boolean type.
+    #[display("bool")]
+    #[serde(rename = "bool")]
+    Boolean,
+    /// A sketch group type.
+    SketchGroup,
+    /// A sketch surface type.
+    SketchSurface,
+    /// An extrude group type.
+    ExtrudeGroup,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, JsonSchema, Bake)]
+#[databake(path = kcl_lib::ast::types)]
+#[serde(tag = "type")]
+pub enum FnArgType {
+    /// A primitive type.
+    Primitive(FnArgPrimitive),
+    // An array of a primitive type.
+    Array(FnArgPrimitive),
```

Comment:
Will this allow T[][][]... ? (KCL supports nested arrays)

Reply:
sure i can do that i dont think we do that in any std lib fn tho

Reply:
on second thought that gets super recursive and until we need it id rather punt, im not that advanced at winnow yet